apps/gallery/js/ImageEditor.js

```diff
@@ -405,23 +439,13 @@ function cancelEditTool() {
       selectEditOption($(savedEditSettings.crop.cropModeId), true);
 
       editSettings.crop = savedEditSettings.crop;
-
-      // In crop edit tool, click of back to original button is different
-      // from other aspect ratio buttons. Click of back to original -
-      // (undoCropHandler) set source rectangle crop region to original image
-      // size and generates new preview. If user cancels after clicking back
-      // to original, we need to explicitly reset preview and set source
-      // rectangle crop region to saved values from editSettings.
-      var isCropped = imageEditor.hasBeenCropped();
-      if (!isCropped) {
-        // if crop region is same as original image size,
-        // cancel out by discarding original preview image.
-        imageEditor.resetPreview();
-        // Set source rectangle crop region from editSettings
-        // to go back to previous cropped size preview.
-        imageEditor.forceSetCropRegion();
-      }
-
+      // Set source rectangle size from saved editSettings
+      // to go back to previous cropped size preview.
+      imageEditor.forceSetCropRegion();
+      // Discard changes and revert to previous crop sizes
+      imageEditor.requestNewPreview(function() {
+        imageEditor.hideCropOverlay();
+      });
```

Comment:
I haven't looked ahead yet to find the definition of requestNewPreview(), but at this point of the code, I don't understand why this case does not need a call to edit() and the other two cases do. What is actually displaying the new preview here?